src/utils/ContainerUtils.ts

```diff
@@ -0,0 +1,89 @@
+                styleObject[name] = pair[1].trim();
+            }
+            return styleObject;
+        }, {});
+    } catch (error) {
+        // tslint:disable-next-line no-console
+        window.console.log("Failed to parse style", style, error);
+    }
+
+    return {};
+};
+export class Utils {
+    static validate(props: ContainerProps & { filterNode: HTMLElement; targetListView: ListView; validate: boolean }): string {
+        const widgetName = "dropdown-filter";
+        // validate filter values if filterby = attribute, then value should not be empty or "" or " ".
+        if (!props.filterNode) {
+            return `${widgetName}: unable to find a listview with to attach to`;
+        }
+
+        if (!(props.targetListView && props.targetListView._datasource)) {
+            return `${widgetName}: this Mendix version is incompatible`;
+        }
+
+        if (props.entity && !Utils.itContains(props.entity, "/")) {
+            if (props.entity !== props.targetListView._entity) {
+                return `${widgetName}: supplied entity "${props.entity}" does not belong to list view data source`;
+            }
+        }
+
+        if (props.filters && !props.filters.length) {
+            return `${widgetName}: should have atleast one filter`;
+        }
+
+        if (props.filters) {
+            const errorMessage: string[] = [];
+            props.filters.forEach((filter, index) => {
+                if (filter.filterBy === "XPath" && !filter.constraint) {
+                    errorMessage.push(`Filter position: {${index + 1 }} is missing XPath constraint`);
+                }
+                if (filter.filterBy === "attribute" && !filter.attributeValue) {
+                    errorMessage.push(`Filter position: {${index + 1 }} is missing a Value constraint`);
+                }
+            });
+            if (errorMessage.length) {
+                return `${widgetName} : ${errorMessage.join(", ")}`;
+            }
+        }
+
+        if (!isNaN(props.defaultFilter) && props.defaultFilter >= 0 && props.defaultFilter > props.filters.length) {
+            return `${widgetName}: Default-Filter value must be less or equal to maximum filter-count '${props.filters.length}'`;
+        }
+
+        return "";
+    }
+
+    static isCompatible(targetListView: ListView): boolean {
+        return !!(targetListView && targetListView._datasource);
+    }
+
+    static findTargetNode(props: ContainerProps, filterNode: HTMLElement): HTMLElement | null {
+        let targetNode: HTMLElement | null = null ;
+
+        while (!targetNode && filterNode) {
+            targetNode = props.targetListViewName
+                ? filterNode.querySelector(`.mx-name-${props.targetListViewName}`) as HTMLElement
+                : filterNode.querySelectorAll(`.mx-listview`)[0] as HTMLElement;
+
+            if (targetNode || filterNode.classList.contains("mx-incubator")
+                || filterNode.classList.contains("mx-offscreen")) {
```

Comment:
Will cause error when no list view is in the page, should check on `!==document` too
as document root will not have a `classList`